vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -27,7 +27,9 @@ def openstack_handle(options = {})
       raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
       username = options[:user] || self.authentication_userid(options[:auth_type])
-      password = options[:pass] || self.authentication_password(options[:auth_type])
+
+      # the |"" is here to prevent an unhelpful ArgumentError from OpenStackHandle::Handle#connect
+      password = options[:pass] || self.authentication_password(options[:auth_type]) || ""
```

Comment:
@himdel Correct, missing_credentials? just checks for userid.

@blomquisg @Fryguy I think the only problem is that some systems are fine with empty passwords.  If we're consistent in accepting empty passwords "" vs. rejecting nil passwords, we should be ok.  But, I'm afraid that some code is doing .blank? which puts "" and nil in the same bucket.

I would think each provider should specify what is considered required credentials.  For some, it could just be a token.  I'd rather not have the authentication stuff do this.  Each provider should say what they require and authentication should provide a means to check for each of these requirements.